### DISCUSSION TITLE
A problem called "Property not_in_nav is not allowed" on VSCode

### DISCUSSION THREAD

**Discussion by CodeZHXS** [squidfunk/mkdocs-material#6129] [Q&A] [answered]
I followed the advice from this [site](https://squidfunk.github.io/mkdocs-material/creating-your-site/?h=vscode#minimal-configuration).
Now the content of my `.vscode/settings.json` is:
```json
{
 ......
    "yaml.schemas": {
      "https://squidfunk.github.io/mkdocs-material/schema.json": "mkdocs.yml",
    },
    "yaml.customTags": [ 
      "!ENV scalar",
      "!ENV sequence",
      "tag:yaml.org,2002:python/name:material.extensions.emoji.to_svg",
      "tag:yaml.org,2002:python/name:material.extensions.emoji.twemoji",
      "tag:yaml.org,2002:python/name:pymdownx.superfences.fence_code_format"
    ]
 ......
 }
```

And the content of my `mkdocs.yml` is:
```yml
......
not_in_nav: |
  index.md
......
```
Then I got a problem from VSCode, "Property not_in_nav is not allowed."

How can it be resolved?

**Comment by squidfunk** [comment]
It's currently not in our JSON schema, but we can and should add the new options for sure – cc @alexvoss

**Reply by CodeZHXS** [reply]
Thx bro!

**Reply by alexvoss** [reply]
@squidfunk I can have a look today.

**Comment by squidfunk** [comment, marked as answer]
Thanks to @alexvoss, our schema now includes the missing MkDocs settings!

**Comment by Kordokareem** [comment]
> I followed the advice from this [site](https://squidfunk.github.io/mkdocs-material/creating-your-site/?h=vscode#minimal-configuration).
> Now the content of my `.vscode/settings.json` is:
> ```json
> {
>  ......
>     "yaml.schemas": {
>       "https://squidfunk.github.io/mkdocs-material/schema.json": "mkdocs.yml",
>     },
>     "yaml.customTags": [ 
>       "!ENV scalar",
>       "!ENV sequence",
>       "tag:yaml.org,2002:python/name:material.extensions.emoji.to_svg",
>       "tag:yaml.org,2002:python/name:material.extensions.emoji.twemoji",
>       "tag:yaml.org,2002:python/name:pymdownx.superfences.fence_code_format"
>     ]
>  ......
>  }
> ```
> 
> And the content of my `mkdocs.yml` is:
> ```yml
> ......
> not_in_nav: |
>   index.md
> ......
> ```
> Then I got a problem from VSCode, "Property not_in_nav is not allowed."
> 
> How can it be resolved?

**Reply by alexvoss** [reply]
I don't think the schema has been pushed out in a release yet. I was traveling and so it took a bit longer and did not make it into the release. Which version are you using? 

One way would be to ignore the warning (it is just that, the `mkdocs.yml` file should work just fine) and wait until the next release that contains today's fixes. 

The other would be to install the latest hottest from the public repo. I don't think the changes have been ported into Insiders yet.

**Comment by squidfunk** [comment]
There was a user posting spam that flooded this issue and our issue tracker. I've deleted the comments and blocked him.
